src/search/search_engines/plugin_astar.cc

```diff
@@ -38,9 +42,14 @@ static shared_ptr<SearchEngine> _parse(OptionParser &parser) {
 
     shared_ptr<eager_search::EagerSearch> engine;
     if (!parser.dry_run()) {
-        auto temp = search_common::create_astar_open_list_factory_and_f_eval(opts);
-        opts.set("open", temp.first);
-        opts.set("f_eval", temp.second);
+        shared_ptr<Evaluator> g = make_shared<g_evaluator::GEvaluator>(
```

Comment:
This change seems unrelated to the issue. Pulling out the computation of "f = g + h" defeats the purpose of a function that specifically creates the open list of A*, in my opinion, i.e., that part is what makes it an A* open list..

Reply:
I split it because I didn't see a good way to return the open list, the f-evaluator and the g-evaluator from the function. So you see a good way to do so? Or should I pass the g-evaluator to the function (removing the need to return it)?

Reply:
I'm working on a different solution that restores the original function and directly inserts the g evaluators into the options object.